src/pages/Accounts/Modals/SendFunds.tsx

```diff
@@ -70,12 +71,43 @@ type AskSendFundsModalProps = {
   onClose(): void
 }
 
+let inThrottle: boolean,
+  lastFn: ReturnType<typeof setTimeout>,
+  lastTime: number;
+const timeout = 500;
+
 export const AskTransferFundsModal: FC<AskSendFundsModalProps> = ({ isVisible, onFinish, senderAddress, onClose }) => {
-  const { accounts } = useAccounts();
+  const { accounts, selectedAccount } = useAccounts();
   const [recipientAddress, setRecipientAddress] = useState<string | Account | undefined>();
   const [amount, setAmount] = useState<string>('');
-  const { chainData } = useApi();
-  const { kusamaFee } = useFee();
+  const { chainData, api } = useApi();
+  const [kusamaFee, setKusamaFee] = useState('0');
+  const getKusamaFee = useCallback(() => {
```

Comment:
Pseudocode:
const [isFeeLoading, setIsFeeLoading] = useState(false);
const getKusamaFee = () => {
// we can ignore throttle and use debounce instead. In that case we will be able to show spinner for fee loading
setIsFeeLoading();
debounce(() => updateFee()) // everytime we recall amountChange -> we will refire updateFeeFunction (debounce)
};
const onAmountChanged = () => {
  ...
  getKusamaFee();
}

Reply:
PROS:
clean code, much shorter, easier to read and maintain
no effects
no global variables outside of component